blockchain/blockchain.go

```diff
@@ -48,6 +48,17 @@ var (
 	supportedStarknetVersion  = semver.MustParse("0.12.0")
 )
 
+type ErrBlockConflictsWithL1 struct {
+	L1HeadHash  *felt.Felt
+	L2BlockHash *felt.Felt
+	Height      uint64
+}
+
+func (e ErrBlockConflictsWithL1) Error() string {
+	return fmt.Sprintf("l2 block hash conflicts with l1 head (height %d): got %s, want %s",
+		e.Height, e.L2BlockHash.String(), e.L1HeadHash.String())
+}
+
```

Comment:
The additional information is not really used by the callee. Let's just make this a string err like `ErrParentDoesNotMatchHead` ?